api_tests/collections/test_views.py

```diff
@@ -1714,3 +1714,296 @@ def test_bulk_delete_link_that_is_not_linked_to_correct_node(self):
         errors = res.json['errors']
         assert_equal(len(errors), 1)
         assert_equal(errors[0]['detail'], 'Node link does not belong to the requested node.')
+
+class TestCollectionRelationshipNodeLinks(ApiTestCase):
+
+    def setUp(self):
+        super(TestCollectionRelationshipNodeLinks, self).setUp()
+        self.user = AuthUserFactory()
+        self.user2 = AuthUserFactory()
+        self.auth = Auth(self.user)
+        self.collection = FolderFactory(creator=self.user)
+        self.admin_node = NodeFactory(creator=self.user)
+        self.contributor_node = NodeFactory(creator=self.user2)
+        self.contributor_node.add_contributor(self.user, auth=Auth(self.user2))
+        self.contributor_node.save()
+        self.other_node = NodeFactory()
+        self.linked_node = NodeFactory(creator=self.user)
+        self.collection.add_pointer(self.linked_node, auth=self.auth)
+        self.public_collection = FolderFactory(is_public=True, creator=self.user2)
+        self.public_collection.add_pointer(self.linked_node, auth=Auth(self.user2))
+        self.public_node = NodeFactory(is_public=True)
+        self.url = '/{}collections/{}/relationships/linked_nodes/'.format(API_BASE, self.collection._id)
+        self.public_url = '/{}collections/{}/relationships/linked_nodes/'.format(API_BASE, self.public_collection._id)
+
+    def payload(self, node_ids=None):
+        node_ids = node_ids or [self.admin_node._id]
+        env_linked_nodes = [{"type": "linked_nodes", "id": node_id} for node_id in node_ids]
+        return {"data": env_linked_nodes}
+
+    def test_get_relationship_linked_nodes(self):
+        res = self.app.get(
+            self.url, auth=self.user.auth
+        )
+
+        assert_equal(res.status_code, 200)
+
+        assert_in(self.collection.linked_nodes_self_url, res.json['links']['self'])
+        assert_in(self.collection.linked_nodes_related_url, res.json['links']['html'])
+        assert_equal(res.json['data'][0]['id'], self.linked_node._id)
+
+    def test_get_public_relationship_linked_nodes_logged_out(self):
+        res = self.app.get(self.public_url)
+
+        assert_equal(res.status_code, 200)
+        assert_equal(res.json['data'][0]['id'], self.linked_node._id)
```

Comment:
For those cases, I only check for the first item because there is only one node being expected. I can change that to be 
```
ids = [data['id'] for data in res.json['data']]
assert_equal(ids, [self.linked_node._id]
```